src/main/java/reposense/model/Author.java

```diff
@@ -139,6 +139,16 @@ public PathMatcher getIgnoreGlobMatcher() {
      * and updates the ignore glob matcher.
      */
     public void appendIgnoreGlobList(List<String> ignoreGlobList) {
+        validateIgnoreGlobs(ignoreGlobList);
+        this.ignoreGlobList.addAll(ignoreGlobList);
+        updateIgnoreGlobMatcher();
+    }
+
+    /**
+     * Validates and adds a list of ignore glob into the {@code Author} class instance variable without duplicates
+     * and updates the ignore glob matcher.
+     */
+    public void appendIgnoreGlobListWithoutDuplicates(List<String> ignoreGlobList) {
```

Comment:
A better name is `setIgnoreGlobList' I believe.

Reply:
If it is truly `appendIgnoreGlobListWithoutDuplicates`, we should use a set instead of removing them all.

Reply:
Now if we have a AuthorConfig with empty branch field, it will map to multiple repos. If the repos have common strings in the IgnoreGlobList, addAll for list will result in duplicates. Can use linkedhashset to replace arraylist without error. Do you wish to change the name from `IgnoreGlobList` to `IgnoreGlobSet` also?

Reply:
Sorry for the late reply.
For me, a pseudo code like below will be enough:
```
if(!list.contains(item)){
    list.add(item)
}
```

Reply:
Sorry for the incovenient caused, can you switch back to the list version?

Reply:
Updated